pre_commit/languages/dotnet.py

```diff
@@ -58,9 +59,15 @@ def install_environment(
         )
 
         # Determine tool from the packaged file <tool_name>.<version>.nupkg
+        # https://learn.microsoft.com/en-us/nuget/concepts/package-versioning#version-basics
+        tool_name_re = re.compile(r'(.*)\.\d+\.\d+\.\d+(-.*)?\.nupkg')
+
         build_outputs = os.listdir(os.path.join(prefix.prefix_dir, build_dir))
         for output in build_outputs:
-            tool_name = output.split('.')[0]
+
+            tool_name_match = tool_name_re.match(output)
+            assert tool_name_match
+            tool_name = tool_name_match.group(1)
```

Comment:
this seems fragile -- can we read the name out of the metadata itself? or install the `.nupkg` file directly?

Reply:
I've committed a version which reads the xml from the nupkg (zip file), but it needs tidied up a bit

Reply:
There might be a way to simplify it, but this version seems to work.